tests/namespace_interceptor_test.go

```diff
@@ -0,0 +1,150 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tests
+
+import (
+	"time"
+
+	"github.com/pborman/uuid"
+
+	commandpb "go.temporal.io/api/command/v1"
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	taskqueuepb "go.temporal.io/api/taskqueue/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/server/common/payloads"
+	"go.temporal.io/server/common/primitives/timestamp"
+)
+
+func (s *functionalSuite) TestServerRejectsInvalidRequests() {
+	sut := newSystemUnderTestConnector(s)
+
+	err := sut.StartWorkflowExecution(s.namespace)
+	s.NoError(err)
+
+	// start workflow task
+	err = sut.PollWorkflowTaskQueue(s.namespace)
+	s.NoError(err)
+
+	err = sut.RespondWorkflowTaskCompleted("another-namespace")
+	s.Errorf(err, "Invalid request was processed, make sure request validator interceptor is loaded")
+
+	err = sut.RespondWorkflowTaskCompleted(s.namespace)
+	s.NoError(err, "Valid request was rejected")
+}
+
+type sutConnector struct {
+	suite           *functionalSuite
+	identity        string
+	taskQueue       *taskqueuepb.TaskQueue
+	stickyTaskQueue *taskqueuepb.TaskQueue
+	id              string
+	taskToken       []byte
+}
+
+func newSystemUnderTestConnector(s *functionalSuite) *sutConnector {
+	id := uuid.New()
+	taskQueue := &taskqueuepb.TaskQueue{Name: id, Kind: enumspb.TASK_QUEUE_KIND_NORMAL}
+	stickyTaskQueue := &taskqueuepb.TaskQueue{Name: "test-sticky-taskqueue", Kind: enumspb.TASK_QUEUE_KIND_STICKY, NormalName: id}
+
+	return &sutConnector{
+		suite:           s,
+		identity:        "worker-1",
+		taskQueue:       taskQueue,
+		stickyTaskQueue: stickyTaskQueue,
+		id:              id,
+	}
+
+}
+
+func (b *sutConnector) StartWorkflowExecution(namespace string) error {
```

Comment:
All these methods don't need to be exported. I.e.:
```suggestion
func (b *sutConnector) startWorkflowExecution(namespace string) error {
```

Reply:
Also I am trying to use special `namespace.Name` and `namespace.ID` types everywhere to pass namespace name and Id.

Reply:
👍